### ISSUE TITLE
Using alias on condition combined with AND or OR fails

### ISSUE THREAD

**Issue by TinchyIzzy** [objectbox/objectbox-java#834]
I want to find (i < 10 or i > 20) list

I have tried 

less(param, 0).or().greater(param, 0)    then  setParameter(param, 10).setParameter(param, 20)
and
less(param, 0).parameterAlias("one").or().greater(param, 0).parameterAlias("two")    then  setParameter("one", 10).setParameter("two", 20)

none of them work

**Comment by greenrobot-team** [comment]
Assuming this is pseudo-code this should be correct. Can you provide a small code example that reproduces this?

**Comment by TinchyIzzy** [comment]
`public class ObjectBoxActivity extends AppCompatActivity {
    private static final String TAG = "ObjectBoxActivity";

    BoxStore store;
    Box<TestEntity> box;
    Query<TestEntity> query;
    Query<TestEntity> query2;
    int size;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.e(TAG, "onCreate: ");

        store = MyObjectBox.builder()
                .androidContext(this)
                .build();
        boolean start = new AndroidObjectBrowser(store).start(this);
        box = store.boxFor(TestEntity.class);
        query = box.query().less(TestEntity_.value, 0).or().greater(TestEntity_.value, 0).build();
//        query2 = box.query().less(TestEntity_.value, 0).parameterAlias("one").or().greater(TestEntity_.value, 0).parameterAlias("two") .build();

        box.removeAll();
        box.put(new TestEntity(9));
        box.put(new TestEntity(7));
        box.put(new TestEntity(6));
        box.put(new TestEntity(16));
        box.put(new TestEntity(13));
        box.put(new TestEntity(23));
        box.put(new TestEntity(26));

        size = box.query().build().find().size();
        Log.e(TAG, "all size: " + size);
        size = query.setParameter(TestEntity_.value, 10).setParameter(TestEntity_.value, 20).find().size();
        Log.e(TAG, "size: " + size);
//        size = query2.setParameter("one", 10).setParameter("two", 20).find().size();
//        Log.e(TAG, "size: " + size);

    }
}`

**Comment by TinchyIzzy** [comment]
> Assuming this is pseudo-code this should be correct. Can you provide a small code example that reproduces this?

`public class ObjectBoxActivity extends AppCompatActivity {
    private static final String TAG = "ObjectBoxActivity";

    BoxStore store;
    Box<TestEntity> box;
    Query<TestEntity> query;
    Query<TestEntity> query2;
    int size;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        Log.e(TAG, "onCreate: ");

        store = MyObjectBox.builder()
                .androidContext(this)
                .build();
        boolean start = new AndroidObjectBrowser(store).start(this);
        box = store.boxFor(TestEntity.class);
        query = box.query().less(TestEntity_.value, 0).or().greater(TestEntity_.value, 0).build();
//        query2 = box.query().less(TestEntity_.value, 0).parameterAlias("one").or().greater(TestEntity_.value, 0).parameterAlias("two") .build();

        box.removeAll();
        box.put(new TestEntity(9));
        box.put(new TestEntity(7));
        box.put(new TestEntity(6));
        box.put(new TestEntity(16));
        box.put(new TestEntity(13));
        box.put(new TestEntity(23));
        box.put(new TestEntity(26));

        size = box.query().build().find().size();
        Log.e(TAG, "all size: " + size);
        size = query.setParameter(TestEntity_.value, 10).setParameter(TestEntity_.value, 20).find().size();
        Log.e(TAG, "size: " + size);
//        size = query2.setParameter("one", 10).setParameter("two", 20).find().size();
//        Log.e(TAG, "size: " + size);

    }
}`

Log
size: 7
size: 7

if open query2
2020-03-11 11:48:31.661 4137-4137/com.xier.irexample E/ObjectBoxActivity: onCreate: 
2020-03-11 11:48:31.676 4137-4137/com.xier.irexample D/AndroidRuntime: Shutting down VM
2020-03-11 11:48:31.676 4137-4137/com.xier.irexample E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.xier.irexample, PID: 4137
    java.lang.RuntimeException: Unable to start activity ComponentInfo{com.xier.irexample/com.xier.irexample.ObjectBoxActivity}: java.lang.IllegalArgumentException: Argument condition "condition->withProperty()" not met (L392)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2671)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2732)
        at android.app.ActivityThread.-wrap12(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1483)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:154)
        at android.app.ActivityThread.main(ActivityThread.java:6141)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:912)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:802)
     Caused by: java.lang.IllegalArgumentException: Argument condition "condition->withProperty()" not met (L392)
        at io.objectbox.query.QueryBuilder.nativeSetParameterAlias(Native Method)
        at io.objectbox.query.QueryBuilder.parameterAlias(QueryBuilder.java:313)
        at com.xier.irexample.ObjectBoxActivity.onCreate(ObjectBoxActivity.java:34)
        at android.app.Activity.performCreate(Activity.java:6709)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1118)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2624)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2732) 
        at android.app.ActivityThread.-wrap12(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1483) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:154) 
        at android.app.ActivityThread.main(ActivityThread.java:6141) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:912) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:802) 




`@Entity
public class TestEntity {
    @Id(assignable = false)
    public long id;
    private int value;

    public TestEntity() {
    }

    public TestEntity(int value) {
        this.value = value;
    }

    public int getValue() {
        return value;
    }

    public void setValue(int value) {
        this.value = value;
    }
}`

**Comment by greenrobot-team** [comment]
Thanks, you have found a bug! Using `parameterAlias(alias)` on a condition combined with `.or()` causes an exception:

```
 Caused by: java.lang.IllegalArgumentException: Argument condition "condition->withProperty()" not met (L392)
    at io.objectbox.query.QueryBuilder.nativeSetParameterAlias(Native Method)
    at io.objectbox.query.QueryBuilder.parameterAlias(QueryBuilder.java:313)
```

Temporary workaround: use [the new Query API](url) in `3.0.0-alpha1`. It does not appear to have this issue as it combines queries differently internally (it creates the conditions first, then combines them).
https://docs.objectbox.io/queries#new-query-api

Example:
```java
Query<TestEntity> query = box.query(
                TestEntity_.value.less(0).alias("less")
                        .or(TestEntity_.value.greater(0).alias("greater"))
        ).build();
```

**Comment by TinchyIzzy** [comment]
When will 3.0.0 be released

**Comment by greenrobot-team** [comment]
When it's done, sorry.

Another workaround I forgot to mention: re-build the `Query` each time.

**Comment by TinchyIzzy** [comment]
3q
I have re-build each time

**Comment by greenrobot-team** [comment]
A fix is available with `3.0.0-alpha2`.

**Comment by greenrobot-team** [comment]
This fix was backported and is now also available with `2.6.0-RC`.
https://docs.objectbox.io/#v-2-6-0-rc-2020-04-28

**Comment by greenrobot-team** [comment]
Fixed with `2.6.0`.
https://docs.objectbox.io/#v-2-6-0-2020-06-09
